javatests/arcs/android/storage/database/DatabaseImplTest.kt

```diff
@@ -165,20 +165,34 @@ class DatabaseImplTest {
     }
 
     @Test
-    fun getStorageKeyId_newKeys() = runBlockingTest {
-        assertThat(database.getEntityStorageKeyId(DummyStorageKey("key1"), 123L, db)).isEqualTo(1L)
-        assertThat(database.getEntityStorageKeyId(DummyStorageKey("key2"), 123L, db)).isEqualTo(2L)
-        assertThat(database.getEntityStorageKeyId(DummyStorageKey("key3"), 123L, db)).isEqualTo(3L)
+    fun getEntityStorageKeyId_newKeys() = runBlockingTest {
+        assertThat(database.getEntityStorageKeyId(DummyStorageKey("key1"), "eid1", 123L, db)).isEqualTo(1L)
+        assertThat(database.getEntityStorageKeyId(DummyStorageKey("key2"), "eid2", 123L, db)).isEqualTo(2L)
+        assertThat(database.getEntityStorageKeyId(DummyStorageKey("key3"), "eid3", 123L, db)).isEqualTo(3L)
     }
 
     @Test
-    fun getStorageKeyId_existingKeys() = runBlockingTest {
+    fun getEntityStorageKeyId_existingKeys() = runBlockingTest {
         // Create new keys.
-        assertThat(database.getEntityStorageKeyId(DummyStorageKey("key1"), 123L, db)).isEqualTo(1L)
-        assertThat(database.getEntityStorageKeyId(DummyStorageKey("key2"), 123L, db)).isEqualTo(2L)
+        assertThat(database.getEntityStorageKeyId(DummyStorageKey("key1"), "eid1", 123L, db)).isEqualTo(1L)
+        assertThat(database.getEntityStorageKeyId(DummyStorageKey("key2"), "eid2", 123L, db)).isEqualTo(2L)
         // Check existing keys.
-        assertThat(database.getEntityStorageKeyId(DummyStorageKey("key1"), 123L, db)).isEqualTo(1L)
-        assertThat(database.getEntityStorageKeyId(DummyStorageKey("key2"), 123L, db)).isEqualTo(2L)
+        assertThat(database.getEntityStorageKeyId(DummyStorageKey("key1"), "eid1", 123L, db)).isEqualTo(1L)
+        assertThat(database.getEntityStorageKeyId(DummyStorageKey("key2"), "eid2", 123L, db)).isEqualTo(2L)
+    }
+
+    @Test
+    fun getEntityStorageKeyId_wrongEntityId() = runBlockingTest {
+        val key = DummyStorageKey("key")
+        database.getEntityStorageKeyId(key, "correct-entity-id", 123L, db)
```

Comment:
Just a question:  the storage key for an entity is its fully-qualified storage key, right? not the collection's backing-store storage key (ie. the base key, where appending the entity-id to it creates the fully-qualified key).

Reply:
That's right. (At least, that's the way I've implemented it)